### PR TITLE
Initial App Import

### PR DESCRIPTION
**Pull request by rajadain**
## Overview

Imports micro site from Model My Watershed. Sets up basic bundling into `/dist`.
### Notes

In an effort to minimize the difference during import, I've tried to keep the scripts consistent with what they were in Model My Watershed. However, they are now inconsistent with all our other projects which run scripts from within the VM, rather than from outside. I'm making cards to address this in the future.

We probably don't need the `--vendor` flag in the bundling script anymore. That is also a holdover from Model My Watershed and should be removed.

Apparently we minify only JavaScript, not any of the CSS. We should look into minifying CSS. Also, I just copied over the entire SASS directory from Model My Watershed. We can probably lose a lot of that.

Maybe we shouldn't be exporting to the `/dist` folder and serving from there for development? This was the easiest thing to do right now, but I feel `/dist` should only be used for production builds.

Even though this is going to be on a CDN, there's still a _lot_ of files being requested. We should look into using CSS rather than Retina.js for serving higher resolution images to reduce the number of requests, and maybe also look into CSS sprites if that is not enough.
## Testing Instructions

Check out this branch. Reload Vagrant so it picks up on new server port mapping. Then run the bundle script:

``` bash
$ ./scripts/bundle.sh --vendor
```

Open [localhost:8000](http://localhost:8000/) in a browser and test that the app works identically to the one on [staging](http://staging.app.wikiwatershed.org/micro), and that there are no errors in the console.

Connects https://github.com/WikiWatershed/model-my-watershed/issues/1483

### CODE REVIEW

**Comment by jtarricone** [comment]
The app behavior looked fine, but it might be notable that `npm install` choked when invoked from `setup.sh`. I have the npm log if it would be of interest.

File: scripts/npm.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+# Pass arguments to npm the project root
+
+set -e
+set -x
+
+ARGS=$*
+
+vagrant ssh -c "cd /vagrant/src/app/ && npm $ARGS"
```
Comment (kdeloach) [inline scripts/npm.sh]: Is this script necessary any longer, considering that we now have `update` and `server`?

File: scripts/server.sh [hunk 2]
```diff
@@ -19,6 +19,6 @@ then
     then
         usage
     else
-      npm run start
+      vagrant ssh -c "cd /vagrant/dist/ && python -m SimpleHTTPServer 8000"
```
Comment (kdeloach) [inline scripts/server.sh]: Is there a reason we need to execute this outside of the VM? The convention we have been following for these scripts (`update`, `server`, etc.) has been to execute them within the context of the VM, so we shouldn't need to wrap these commands with `vagrant ssh ...`, right?

File: src/app/package.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+      "watchify": "3.1.0"
+  },
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/WikiWatershed/mmw-micro.git"
+  },
+  "author": "Azavea Inc.",
```
Comment (kdeloach) [inline src/app/package.json]: Consider changing author to just "Azavea".

File: scripts/cibuild.sh [hunk 4]
```diff
@@ -21,6 +21,6 @@ then
     else
         ./scripts/update.sh
         ./scripts/test.sh
-        npm run bundle
+        ./src/app/bundle.sh --vendor --minify
```
Comment (kdeloach) [inline scripts/cibuild.sh]: I think the bundle script should be called from `update`. Maybe we can use a "production" environment variable to indicate that minification needs to happen.

**Comment by kdeloach** [comment]
Can we change this so that running `./scripts/update.sh; ./scripts/server.sh` works as expected, instead of having to run `./scripts/bundle.sh --vendor`?

**Comment by rajadain** [comment]
Updated according to all feedback.

Starting with a fresh checkout of the repo, `./scripts/setup.sh` should finish without any errors. All other scripts are expected to be run from within Vagrant.

File: scripts/npm.sh (hunk 1 above)
Comment (rajadain) [reply scripts/npm.sh]: Removed `npm.sh`

File: scripts/server.sh (hunk 2 above)
Comment (rajadain) [reply scripts/server.sh]: Removed `vagrant ssh` wrapper, this is now expected to be run from within.

**Comment by kdeloach** [comment]
+1

File: README.md [hunk 5]
```diff
@@ -10,13 +10,12 @@ A static micro site storm model.
...
-Clone the project, `cd` into the directory, then run `./scripts/setup.sh` to create the Vagrant VM and then build the Docker containers.
-
-To start the servers during development:
+Clone the project, `cd` into the directory, then run `./scripts/setup.sh` to create the Vagrant VM and install npm dependencies.
 
+To start the server during development:
 ```bash
 $ vagrant ssh
-$ ./scripts/server.sh
+$ ./scripts/server.ssh
```
Comment (kdeloach) [inline README.md]: Typo on this line.

Comment (rajadain) [reply README.md]: Thanks, fixed.

**Pull request merged**